File: modules/0104_exploratory_data_analysis/10_data_visualisation_in_Python/script_04.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Thu Oct  5 14:16:25 2023

@author: jerry
"""


import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import calendar


temp_data = pd.read_csv("../../../data/eda/data_visualisation/Average Temperatures in NY.csv")
temp_data.info()


temp_agg = pd.pivot_table(temp_data, index = ['Month '], columns = ['Year '])
temp_agg.columns = temp_data['Year '].unique()
temp_agg = temp_agg.reindex(calendar.month_abbr)


ax = sns.heatmap(temp_agg)
ax.set(title = 'Heatmap', xlabel = 'Year', ylabel = 'Age Group')

File: src/0104_exploratory_data_analysis/05_data_visualisation/script_04.R

library(plotly)

heatmap_data <- read.csv("./data/0104_exploratory_data_analysis/data_visualisation/Average\ Temperatures\ in\ NY.csv", header = TRUE)
heatmap_data$Month <- factor(heatmap_data$Month, level = unique(heatmap_data$Month))

plot_ly(
  heatmap_data, 
  x = heatmap_data$Month, 
  y = heatmap_data$Year, 
  z = heatmap_data$Temperature, 
  type = "heatmap",
  connectgaps = FALSE,
  showscale = TRUE
)
